shark-graph/src/main/java/shark/internal/FieldValuesReader.kt

```diff
@@ -44,7 +44,7 @@ internal class FieldValuesReader(
     }
   }
 
-  private fun readId(): Long {
+  fun readId(): Long {
```

Comment:
Made this public to avoid redundant check `when (field.type) == PrimitiveType.REFERENCE_HPROF_TYPE` that comes in `readValue` method. 

Decompiled version of that method contains additional null check (which we should remove for the whole library, we don't yet)
```
 @NotNull
   public final ValueHolder readValue(@NotNull FieldRecord field) {
      Intrinsics.checkParameterIsNotNull(field, "field");
      int var2 = field.getType();
      ValueHolder var10000;
      if (var2 == 2) {
         var10000 = (ValueHolder)(new ReferenceHolder(this.readId()));
```

Reply:
`FieldValuesReader` is internal so no new API published, all good.